pubpub/platform
# Create pubs from forms

## Issue(s) Resolved
Closes https://github.com/pubpub/platform/issues/669

## High-level Explanation of PR
On a page like `/croccroc/public/forms/{form-slug}/fill`, you no longer need the `pubId` query parameter. Now, if there is no `pubId`, the form will have a "create" behavior. Instead of querying for an existing pub, we will create a pub.

## Test Plan
* Create a form
* Visit the form at `/{community}/public/forms/{form-slug}/fill`
* You should see a blank form and you should be able to submit it **once you fill all fields out** (see note)
* Once submitted, you should see the post-submission behavior (and if you used any pub specific markdown, it should still render ok)
* You should be able to see created pubs

### Other things to test
* File upload works and uploads to the proper pub ID
* Configure submit button with
  * a stage (the pub should be created in this stage)
  * special markdown, i.e. with `:value{field="croccroc:title"}`

## Screenshots (if applicable)

### Filling out a form without a pub id

https://github.com/user-attachments/assets/11c09b07-9a49-4c11-adb7-96e68054429c

### Copying link to form

https://github.com/user-attachments/assets/f2dc9032-df6f-4f70-8436-037c5718bfa1

## Notes
Right now the form lets you submit even if not all fields are filled out, but there will be an error because the pub fails validation on create. I think we should come back to this after https://github.com/pubpub/platform/pull/685, which might make it not a problem at all!
